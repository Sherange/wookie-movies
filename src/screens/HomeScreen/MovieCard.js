import React from 'react';
import {Image, StyleSheet, Text, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';
import {primaryTextColor} from '../../constants/theme';

const MovieCard = props => {
  const {poster, title, length, classification} = props.item;
  return (
    <TouchableOpacity
      style={styles.movieCardContainer}
      onPress={() => props.navigateDetailScreen(props.item)}>
      <Image source={{uri: poster}} style={{width: 150, height: 250}}></Image>
      <Text style={styles.titleStyle} numberOfLines={1}>
        {title}
      </Text>
      <Text style={styles.subTitleStyle} numberOfLines={1}>
        {length} | {classification}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  movieCardContainer: {
    width: 150,
    marginRight: 10,
    marginVertical: 8,
  },
  titleStyle: {
    marginVertical: 4,
    color: primaryTextColor,
    fontSize: 16,
    fontWeight: '600',
  },
  subTitleStyle: {
    marginVertical: 2,
    color: primaryTextColor,
    opacity: 0.6,
    fontSize: 14,
  },
});

MovieCard.propTypes = {
  item: PropTypes.shape({
    poster: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    length: PropTypes.string.isRequired,
    classification: PropTypes.string.isRequired,
  }),
  navigateDetailScreen: PropTypes.func.isRequired,
};

export default MovieCard;
