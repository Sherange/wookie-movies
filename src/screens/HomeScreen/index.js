import React from 'react';
import {SafeAreaView, View, StatusBar, Text, StyleSheet} from 'react-native';
import {backgroundColor, primaryTextColor} from '../../constants/theme';

const HomeScreen = () => {
  return (
    <SafeAreaView style={styles.safeAreaViewStyle}>
      <View style={styles.homeScreen}>
        <StatusBar barStyle={'light-content'} translucent={true} />

        <Text style={styles.titleStyle}>Home Screen</Text>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  safeAreaViewStyle: {
    flex: 1,
    backgroundColor: backgroundColor,
  },
  homeScreen: {
    flex: 1,
    backgroundColor: backgroundColor,
    justifyContent: 'center',
    alignItems: 'center',
  },
  titleStyle: {
    color: primaryTextColor,
    fontSize: 24,
    fontWeight: '600',
    marginLeft: 10,
  },
});

export default HomeScreen;
