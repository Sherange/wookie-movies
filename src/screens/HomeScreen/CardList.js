import React from 'react';
import {View, Text, StyleSheet, FlatList} from 'react-native';
import {primaryTextColor, secondryColor, secondryTextColor} from "../../constants/theme";
import MovieCard from './MovieCard';
import PropTypes from 'prop-types';

const CardList = props => {
  const {data, genre} = props;

  const renderItem = ({item}) => (
    <MovieCard
      image={item.poster}
      title={item.title}
      classification={item.classification}
      length={item.length}
    />
  );
  return (
    <View style={styles.cardListContainer}>
      <Text style={styles.genreTitleText}>{genre}</Text>
      <View style={styles.cardListRow}>
        <FlatList
          data={data}
          horizontal={true}
          showsHorizontalScrollIndicator={false}
          renderItem={renderItem}
          keyExtractor={item => item.id}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  cardListContainer: {
    flex: 1,
    justifyContent: 'center',
    marginHorizontal: 20,
    marginVertical: 10,
  },
  cardListRow: {
    flexDirection: 'row',
  },
  genreTitleText: {
    color: secondryColor,
  },
});

CardList.propTypes = {
  data: PropTypes.array.isRequired,
  genre: PropTypes.string.isRequired,
};

export default CardList;
