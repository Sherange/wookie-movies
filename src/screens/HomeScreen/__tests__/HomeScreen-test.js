import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../index';

it('renders correctly', () => {
  renderer.create(<HomeScreen />);
});

test('Home snapshot', () => {
  const snapshot = renderer.create(<HomeScreen />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
