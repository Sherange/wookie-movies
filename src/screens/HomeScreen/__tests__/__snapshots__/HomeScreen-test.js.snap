// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#0F1B2B",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0F1B2B",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "fontWeight": "600",
          "marginLeft": 10,
        }
      }
    >
      Home Screen
    </Text>
  </View>
</RCTSafeAreaView>
`;
